Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -1094,7 +1094,42 @@ def checkout(self, argument, prune=None):
                 cwd=self.root_path,
             ).returncode else self.commit()
 
+        match = self.dev_branches.match(argument)
         branch_remote = self.remote_for(argument)
+
+        # Branch is not dev and exists on a remote.
+        if not match and branch_remote:
+            path = '{}/{}'.format(branch_remote, argument)
+            local_head = self.commit(branch=argument, include_log=False, include_identifier=False)
+            remote_head = self.commit(branch=path, include_log=False, include_identifier=False)
+            local_bp = self.branch_point(ref=local_head.hash)
+            merge_base = run(
+                [self.executable(), 'merge-base', local_bp.hash, remote_head.hash],
+                cwd=self.root_path,
+                capture_output=True,
+                encoding='utf-8',
+            ).stdout.strip()
+
+            # Resets branch if local is not tracking force pushed remote.
+            if merge_base != local_bp.hash:
+                if local_bp.hash != local_head.hash:
+                    log.info(" You have unsaved changes on the local branch.")
+                    if prompt:
```

Comment:
You can combine this pair into a single `if prompt and Terminal.choose(...` to be less verbose.